paper-server/src/main/java/org/bukkit/craftbukkit/generator/CraftChunkData.java

```diff
@@ -180,4 +181,13 @@ private void setBlock(int x, int y, int z, BlockState type) {
             access.removeBlockEntity(blockPosition);
         }
     }
+
+    @Override
+    public int getHeight(HeightMap heightMap, final int x, final int z) {
+        Preconditions.checkArgument(heightMap != null, "HeightMap cannot be null");
+        if (x < 0 || x > 15 || z < 0 || z > 15) {
+            throw new IllegalArgumentException("Cannot get height outside of a chunks bounds");
```

Comment:
maybe print the coords, just so it's easier to see what has happened when debugging, etc; People mistaking block/chunk bound coords is always kinda fun

Reply:
Ah yeah, fair point... will change now!